devito/passes/iet/mpi.py

```diff
@@ -39,7 +39,13 @@ def _drop_halospots(iet):
 
     # If a HaloSpot Dimension turns out to be SEQUENTIAL, then the HaloSpot is useless
     for hs, iterations in MapNodes(HaloSpot, Iteration).visit(iet).items():
-        if any(i.is_Sequential for i in iterations if i.dim.root in hs.dimensions):
+        dmapper = as_mapper(iterations, lambda i: i.dim.root)
+        flag = False
+        for d, v in dmapper.items():
+            if d in hs.dimensions and all(i.is_Sequential for i in v):
+                flag = True
+                break
+        if flag:
```

Comment:
Do you need that flag?

```
for d, v in dmapper.items():
            if d in hs.dimensions and all(i.is_Sequential for i in v):
                mapper[hs].update(set(hs.functions))
                break
```

doesn't this do the same?

Reply:
oh yes ! I'm an idiot -- basically some leftover from some more complicated code I had before. Just running tests locally with your version. Will push the update if all remains green (as it should!)